tests/RepositoryForms/FieldType/Mapper/SelectionFormMapperTest.php

```diff
@@ -0,0 +1,154 @@
+        $fieldForm->expects($this->once())
+            ->method('getConfig')
+            ->willReturn($config);
+
+        $data = $this->getMockBuilder(\EzSystems\RepositoryForms\Data\Content\FieldData::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $fieldDefinition = $this->getMockBuilder(\eZ\Publish\API\Repository\Values\ContentType\FieldDefinition::class)
+            ->getMock();
+        $fieldDefinition->expects($this->once())
+            ->method('getName')
+            ->willReturn(false);
+        $fieldDefinition->expects($this->once())
+            ->method('getNames')
+            ->willReturn(['foo']);
+        $fieldDefinition
+            ->expects($this->any())
+            ->method('__get')
+            ->will(
+                $this->returnValueMap([
+                    ['isRequired', false],
+                    ['fieldSettings', ['isMultiple' => false, 'options' => []]],
+                ])
+            );
+
+        $data->expects($this->atLeastOnce())
+            ->method('__get')
+            ->with('fieldDefinition')
+            ->willReturn($fieldDefinition);
+
+        $mapper->mapFieldValueForm($fieldForm, $data);
+    }
+
+    public function testMapFieldValueFormWithLanguageCode()
+    {
+        $mapper = new SelectionFormMapper($this->fieldTypeService);
+
+        $config = $this->getMockBuilder(\Symfony\Component\Form\FormConfigInterface::class)->getMock();
+        $config->expects($this->once())
+            ->method('getOption')
+            ->with('languageCode')
+            ->willReturn('eng-GB');
+
+        $formFactory = $this->getMockBuilder(\Symfony\Component\Form\FormFactoryInterface::class)
+            ->setMethods(['addModelTransformer', 'setAutoInitialize', 'getForm'])
+            ->getMockForAbstractClass();
+        $formFactory->expects($this->once())
+            ->method('createBuilder')
+            ->willReturn($formFactory);
+        $formFactory->expects($this->once())
+            ->method('create')
+            ->willReturn($formFactory);
+        $formFactory->expects($this->once())
+            ->method('addModelTransformer')
+            ->willReturn($formFactory);
+        $formFactory->expects($this->once())
+            ->method('setAutoInitialize')
+            ->willReturn($formFactory);
+
+        $config->expects($this->once())
+            ->method('getFormFactory')
+            ->willReturn($formFactory);
+
+        $fieldForm = $this->getMockBuilder(\Symfony\Component\Form\FormInterface::class)->getMock();
+        $fieldForm->expects($this->once())
+            ->method('getConfig')
+            ->willReturn($config);
+
+        $data = $this->getMockBuilder(\EzSystems\RepositoryForms\Data\Content\FieldData::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $fieldDefinition = $this->getMockBuilder(\eZ\Publish\API\Repository\Values\ContentType\FieldDefinition::class)
```

Comment:
In general, I would avoid mocking value objects, and would just create instances. It would greatly shorten and simplify the unit test.